pkg/services/accesscontrol/resourcepermissions/store.go

```diff
@@ -703,3 +735,78 @@ func managedPermission(action, resource string, resourceID, resourceAttribute st
 		Scope:  accesscontrol.Scope(resource, resourceAttribute, resourceID),
 	}
 }
+
+/*
+ACTION SETS
+Stores actionsets IN MEMORY
+*/
+// - To grant access to users/teams/basic roles during runtime, users will go through the managed permissions view and access-control will be enforced based on their permissions.
+// ActionSet is a struct that represents a set of actions that can be performed on a resource.
+// An example of an action set is "folders:edit" which represents the actions that can be performed on a datasource with the uid of 1.
+/*
+```go
+actionSet := &ActionSet{
+	Resource:   "folders",
+	Permission: "edit",
+	Actions:    []string{"folders:read", "folders:write", "dashboards:read", "dashboards:write"},
+}`
+*/
+type ActionSet struct {
+	Action     string   `json:"action"`
+	Resource   string   `json:"resource"`
+	Permission string   `json:"permission"`
+	Actions    []string `json:"actions"`
+}
+
+// InMemoryActionSets is an in-memory implementation of the ActionSetService.
+type InMemoryActionSets struct {
+	log        log.Logger
+	actionSets map[string]*ActionSet
+}
+
+// NewInMemoryActionSets returns a new instance of InMemoryActionSetService.
+func NewInMemoryActionSets(log log.Logger) *InMemoryActionSets {
+	return &InMemoryActionSets{
+		actionSets: make(map[string]*ActionSet),
+		log:        log,
+	}
+}
+
+// GetActionSet returns the action set for the given action.
+func (s *InMemoryActionSets) GetActionSet(actionName string) *ActionSet {
+	actionSet, ok := s.actionSets[actionName]
+	if !ok {
+		return nil
+	}
+	return actionSet
+}
+
+// CreateActionSet function creates an action set from a list of actions and stores it inmemory.
+func (s *InMemoryActionSets) CreateActionSet(resource, permission string, actions []string) (*ActionSet, error) {
```

Comment:
```suggestion
func (s *InMemoryActionSets) GetActionSetName(resource, permission string) (string, error) {
```

refactor function to take in the fixed:X
actions for example
https://github.com/grafana/grafana/blob/main/pkg/services/accesscontrol/ossaccesscontrol/permissions_services.go#L221-L233